### PR TITLE
Updated filtered list test cases for unmark order and mark order

### LINKED ISSUE #129: Fix unused test case in DeleteOrderCommandTest

### LINKED ISSUE #146: Add test case for filtered order list in mark order command test

### LINKED ISSUE #147: Add test case for filtered order list in unmark command test case

### LINKED ISSUE #159: Refactor Status.isCompleted() method

### PR DESCRIPTION
**Pull request by yu-sutong**
close #129 
close #146 
close #147 
close #159

### CODE REVIEW

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java [hunk 1]
```diff
@@ -20,41 +22,65 @@
...
-    public void execute_validIndexOrderList_success() {
+    public void execute_validIndexUnfilteredOrderList_success() {
         MarkOrderCommand markOrderCommand = new MarkOrderCommand(INDEX_FIRST_ORDER);
 
         Model expectedModel = model.createCopy();
-        Order orderToBeMarked = expectedModel.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased())
-                .getOrderList().get(INDEX_FIRST_ORDER.getZeroBased());
-        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarked);
+        Order orderToBeMarkedInUnfilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarkedInUnfilteredList);
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java:47]: Why u using createMarkedOrder to test your execute? It itself is part of execution process and u have no individual test for it to prove it works.

Comment (FooChao) [reply src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java:47]: Consider using order builder instead.

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java [hunk 2]
```diff
@@ -20,41 +22,65 @@
...
-        Person person = model.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased());
-        model.updateSelectedPerson(person);
-        Index outOfBoundIndex = Index.fromOneBased(person.getOrderList().size() + 1);
+    public void execute_validIndexFilteredOrderList_success() {
+        personToMarkOrderUnder.updateFilteredOrderList(new StatusEqualsKeywordPredicate(Status.PENDING));
+        MarkOrderCommand markOrderCommand = new MarkOrderCommand(INDEX_FIRST_ORDER);
+
+        Model expectedModel = model.createCopy();
+        Order orderToBeMarkedInFilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarkedInFilteredList);
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java:63]: Same as above

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java [hunk 3]
```diff
@@ -19,41 +21,67 @@
...
-        Person person = model.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased());
-        model.updateSelectedPerson(person);
-        Index outOfBoundIndex = Index.fromOneBased(person.getOrderList().size() + 1);
+    public void execute_validIndexFilteredOrderList_success() {
+        personToUnmarkOrderUnder.updateFilteredOrderList(new StatusEqualsKeywordPredicate(Status.COMPLETED));
+        UnmarkOrderCommand unmarkOrderCommand = new UnmarkOrderCommand(INDEX_FIRST_ORDER);
+
+        Model expectedModel = model.createCopy();
+        Order orderToBeUnmarkedInUnfilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        Order unmarkedOrder = UnmarkOrderCommand.createUnmarkedOrder(orderToBeUnmarkedInUnfilteredList);
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java:63]: See above

**Review by FooChao — COMMENTED** [review]
Just some small nitpicks. I reviewing on my phone will look at it again tmr.

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java [hunk 4]
```diff
@@ -20,41 +22,65 @@
...
+    public void execute_invalidIndexUnfilteredOrderList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
+        MarkOrderCommand markOrderCommand = new MarkOrderCommand(outOfBoundIndex);
+
+        assertCommandFailure(markOrderCommand, model, Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredOrderList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java:82]: the whole point of testing filtered list is to test something that is out of bounds for filtered but not out of bounds before filtered. If not its purpose will be same as that of unfiltered out of bounds. 
maybe you can either copy from me or the ab3?
image
image

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/OrderCommandTestUtil.java [hunk 5]
```diff
@@ -118,4 +120,18 @@ public static void showOrderAtIndex(Model model, Index targetIndex) {
...
+
+    /**
+     * Returns the order in the model's selected filtered order list by the given {@code Index}.
+     */
+    public static Order getOrderByIndex(Model model, Index index) {
+        return model.getFilteredOrderList().get(index.getZeroBased());
+    }
+
+    /**
+     * Returns the order list in the model's selected person's filtered order list by the given {@code Index}.
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/OrderCommandTestUtil.java:132]: order list in selected person's filtered order list?
How can an order list be in another order list?

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java [hunk 6]
```diff
@@ -19,41 +21,67 @@
...
-    public void execute_validIndexOrderList_success() {
+    public void execute_validIndexUnfilteredOrderList_success() {
         UnmarkOrderCommand unmarkOrderCommand = new UnmarkOrderCommand(INDEX_SECOND_ORDER);
 
         Model expectedModel = model.createCopy();
-        Order orderToBeUnmarked = expectedModel.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased())
-                .getOrderList().get(INDEX_SECOND_ORDER.getZeroBased());
-        Order unmarkedOrder = UnmarkOrderCommand.createUnmarkedOrder(orderToBeUnmarked);
+        Order orderToBeUnmarkedInUnfilteredList = getOrderByIndex(expectedModel, INDEX_SECOND_ORDER);
+        Order unmarkedOrder = UnmarkOrderCommand.createUnmarkedOrder(orderToBeUnmarkedInUnfilteredList);
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java:46]: see previous review

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java [hunk 7]
```diff
@@ -19,41 +21,67 @@
...
+    public void execute_invalidIndexUnfilteredOrderList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
+        UnmarkOrderCommand unmarkOrderCommand = new UnmarkOrderCommand(outOfBoundIndex);
+
+        assertCommandFailure(unmarkOrderCommand, model, Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredOrderList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredOrderList().size() + 1);
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java:83]: see above

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/OrderCommandTestUtil.java (hunk 5 above)
Comment (kaixuan477) [reply src/test/java/seedu/sellsavvy/logic/commands/ordercommands/OrderCommandTestUtil.java:132]:
```suggestion
     * Returns an {@code OrderList} of the model's selected person by the given {@code Index}.
```

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java [hunk 8]
```diff
@@ -20,41 +22,72 @@
...
+    public void execute_validIndexUnfilteredOrderList_success() {
         MarkOrderCommand markOrderCommand = new MarkOrderCommand(INDEX_FIRST_ORDER);
 
         Model expectedModel = model.createCopy();
-        Order orderToBeMarked = expectedModel.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased())
-                .getOrderList().get(INDEX_FIRST_ORDER.getZeroBased());
-        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarked);
+        Order orderToBeMarkedInUnfilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        OrderBuilder orderBuilder = new OrderBuilder(orderToBeMarkedInUnfilteredList);
+        Order markedOrder = orderBuilder.withStatus(Status.COMPLETED).build();
```
Comment (kaixuan477) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java:49]: Order builder can be used as new OrderBuilder().withStatus() iirc, for reference see typicalOrders

**Review by kaixuan477 — COMMENTED** [review]
Small nitpick, looks good. Let the rest review first.

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/OrderCommandTestUtil.java [hunk 9]
```diff
@@ -118,4 +120,18 @@ public static void showOrderAtIndex(Model model, Index targetIndex) {
 
         assertEquals(1, model.getFilteredPersonList().size());
     }
+
+    /**
+     * Returns the order in the model's selected filtered order list by the given {@code Index}.
```
Comment (kaixuan477) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/OrderCommandTestUtil.java]:
```suggestion
     * Returns an {@code Order} from the model's filtered order list by the given {@code Index}.
```

File: src/main/java/seedu/sellsavvy/model/ModelManager.java [hunk 10]
```diff
@@ -155,12 +155,12 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
...
     @Override
     public ReadOnlyObjectProperty<Person> getSelectedPersonProperty() {
-        return selectedPersonPropery.getReadOnlyProperty();
+        return selectedPersonProperty.getReadOnlyProperty();
     }
 
     @Override
     public void updateSelectedPerson(Person person) {
-        selectedPersonPropery.set(person);
+        selectedPersonProperty.set(person);
```
Comment (FooChao) [inline src/main/java/seedu/sellsavvy/model/ModelManager.java:163]: Nice!

**Review by FooChao — APPROVED** [review]
LGTM. But check if they want see and maybe fix what kx say? (idk what he saying but it is for u to figure it out)

**Pull request merged**